src/component/panels/RangesTablePanel.jsx

```diff
@@ -12,21 +12,39 @@ import { useChartData } from '../context/ChartContext';
 import NoTableData from './placeholder/NoTableData';
 
 const RangesTablePanel = () => {
-  //   const [ranges, setRanges] = useState([]);
-  const [ranges] = useState([]); // to replace by line above
+  const [ranges, setRanges] = useState([]);
   const { activeSpectrum, data } = useChartData();
 
-  return activeSpectrum && data && ranges.length > 0 ? (
+  useEffect(() => {
```

Comment:
there are no side effects in this function. you can remove the useState and useEffect and directly write the code in the component.